### PR TITLE
Crud preprocess

### PR DESCRIPTION
**Pull request by azime**

### CODE REVIEW

File: tests/integration/mongo/preprocesses_api_test.py [hunk 1]
```diff
@@ -0,0 +1,145 @@
...
+            "bano_data": {"key": "data_sources_id", "value": "bano_75"}
+        }
+    }]
+    raw = post(app, '/contributors', json.dumps(post_data))
+    r = to_json(raw)
+    assert raw.status_code == 201, print(r)
+
+    preprocess_id = r["contributors"][0]["preprocesses"][0]["id"]
+    raw = delete(app, '/contributors/id_test/preprocesses/{}'.format(preprocess_id))
+    assert raw.status_code == 204, print(to_json(raw))
```
Comment (kadhikari) [inline tests/integration/mongo/preprocesses_api_test.py]: Could you verify the result of get(app, '/contributors/id_test/preprocesses/) which should be 0

File: tests/integration/mongo/preprocesses_api_test.py [hunk 2]
```diff
@@ -0,0 +1,145 @@
...
+    }
+
+    raw = patch(app, '/contributors/id_test/preprocesses/{}'.format(preprocess_id), json.dumps(new_preprocess))
+    r = to_json(raw)
+    assert raw.status_code == 200, print(r)
+    assert len(r["preprocesses"]) == 1
+    assert r["preprocesses"][0]["type"] == new_preprocess["type"]
+    assert r["preprocesses"][0]["source_params"] == new_preprocess["source_params"]
+
+
```
Comment (kadhikari) [inline tests/integration/mongo/preprocesses_api_test.py]: We can also add a new test on patch with two preprocesses of a contributor and te patch modifies only one?

File: tartare/interfaces/preprocess.py [hunk 3]
```diff
@@ -0,0 +1,96 @@
...
+
+    def delete(self, contributor_id, preprocess_id):
+        try:
+            nb_deleted = models.PreProcess.delete(contributor_id, preprocess_id)
+            if nb_deleted == 0:
+                raise ObjectNotFound("Preprocess '{}' not found.".format(contributor_id))
+        except ValueError as e:
+            raise InvalidArguments(str(e))
+
+        return {'preprocesses': []}, 204
```
Comment (datanel) [inline tartare/interfaces/preprocess.py]: new blank line

File: tests/integration/mongo/preprocesses_api_test.py [hunk 4]
```diff
@@ -0,0 +1,197 @@
...
+        "source_params": {
+            "tc_data": {"key": "data_sources.data_format", "value": "gtfs"}
+        }
+    }
+
+    raw = patch(app, '/contributors/id_test/preprocesses/{}'.format(preprocess_id), json.dumps(new_preprocess))
+    r = to_json(raw)
+    assert raw.status_code == 400, print(r)
+    assert 'error' in r
+    assert r['error'] == 'Invalid process type bob'
```
Comment (datanel) [inline tests/integration/mongo/preprocesses_api_test.py]: new blank line

File: tests/integration/mongo/preprocesses_api_test.py (hunk 1 above)
Comment (azime) [inline tests/integration/mongo/preprocesses_api_test.py]: ok

File: tests/integration/mongo/preprocesses_api_test.py (hunk 2 above)
Comment (azime) [inline tests/integration/mongo/preprocesses_api_test.py]: ok

File: tartare/interfaces/preprocess.py (hunk 3 above)
Comment (azime) [inline tartare/interfaces/preprocess.py]: ok

File: tests/integration/mongo/preprocesses_api_test.py (hunk 4 above)
Comment (azime) [inline tests/integration/mongo/preprocesses_api_test.py]: ok

**Pull request merged**